synapse/rest/client/v2_alpha/devices.py

```diff
@@ -70,11 +72,15 @@ def on_GET(self, request, device_id):
 
     @defer.inlineCallbacks
     def on_DELETE(self, request, device_id):
-        # XXX: it's not completely obvious we want to expose this endpoint.
-        # It allows the client to delete access tokens, which feels like a
-        # thing which merits extra auth. But if we want to do the interactive-
-        # auth dance, we should really make it possible to delete more than one
-        # device at a time.
+        body = servlet.parse_json_object_from_request(request)
+
+        authed, result, params, _ = yield self.auth_handler.check_auth([
+            [constants.LoginType.PASSWORD],
```

Comment:
Hmm, this is probably fine for now, but isn't going to work with servers that only allow e.g. LDAP auth.